lib/rubocop/cop/rspec/cop.rb

```diff
@@ -41,32 +33,28 @@ def relevant_file?(file)
         private
 
         def relevant_rubocop_rspec_file?(file)
-          rspec_pattern =~ file
+          self.class.rspec_pattern =~ file
         end
 
-        def rspec_pattern
-          if rspec_pattern_config?
-            Regexp.union(rspec_pattern_config.map(&Regexp.public_method(:new)))
-          else
-            DEFAULT_PATTERN_RE
+        class << self
+          def rspec_pattern
+            @rspec_pattern ||=
+              Regexp.union(
+                rspec_pattern_config.map(&Regexp.public_method(:new))
+              )
           end
-        end
 
-        def all_cops_config
-          config
-            .for_all_cops
-        end
+          private
 
-        def rspec_pattern_config?
-          return unless all_cops_config.key?('RSpec')
+          def rspec_pattern_config
+            default_configuration =
+              RuboCop::RSpec::CONFIG.fetch('AllCops').fetch('RSpec')
 
-          all_cops_config.fetch('RSpec').key?('Patterns')
-        end
-
-        def rspec_pattern_config
-          all_cops_config
-            .fetch('RSpec', DEFAULT_CONFIGURATION)
-            .fetch('Patterns')
+            Config.new
+              .for_all_cops
+              .fetch('RSpec', default_configuration)
+              .fetch('Patterns')
```

Comment:
Isn't the configuration merged with the default configuration, can we assume that `AllCops/RSpec/Patterns` is always present?

Reply:
You are right. We can just use `ConfigLoader.default_configuration.for_all_cops` here.

Reply:
Still, I’m struggling to figure out how to make this work with the changing configurations in cop_spec.rb

Reply:
Just found this https://github.com/rubocop-hq/rubocop/blob/8842e13621faa82c2688e6d5e7cd891e0aa5873c/lib/rubocop/rspec/shared_contexts.rb#L76
Maybe can be of some help?

Reply:
I mean it could be part of 2.0, not even post-2.0. We have a bunch of pull requests specifically for 2.0.